TF_Tooling_Web/src/Connection/MessageReceiver.ts

```diff
@@ -152,16 +152,20 @@ export class MessageReceiver {
         }
     }
 
+    lastKnownCursorPosition: Array<number> = [0,0];
+
     // Function: CheckForAction
     // Checks <actionQueue> for valid <TouchFreeInputActions>. If there are too many in the queue,
     // clears out non-essential <TouchFreeInputActions> down to the number specified by
     // <actionCullToCount>. If any remain, sends the oldest <TouchFreeInputAction> to
     // <InputActionManager> to handle the action.
+    // UP <InputType>s have their positions set to the last known position to ensure
+    // input events trigger correctly.
     CheckForAction(): void {
         while (this.actionQueue.length > this.actionCullToCount) {
             if (this.actionQueue[0] !== undefined) {
                 // Stop shrinking the queue if we have a 'key' input event
-                if (this.actionQueue[0].InteractionFlags & InputType.MOVE) {
+                if ((this.actionQueue[0].InteractionFlags & InputType.MOVE) != 0) {
```

Comment:
This change seems unrelated to the cursor position. What is it for?

Reply:
That's my bad, I thought the bit comparison was acting oddly so I added an explicit check but it wasn't and forgot to remove. Will revert